vignettes/04-synthesis-data.Rmd

```diff
@@ -1 +1,216 @@
+The `inflection` R package is used for the calculation. 
+The resulting inflection points for each cultivar are plotted as a histogram. 
+
+```{r calculate_inflections}
+inflection_points <- c(cultivar = as.character(), value = as.numeric())
+for(each_cultivar in unique(all_cultivars$cultivar)){
+  each_cultivar_df  <- filter(all_cultivars, cultivar == each_cultivar)
+  each_cultivar_inf <- ede(each_cultivar_df$gdd_cum, each_cultivar_df$mean_predict, 0)
+  each_cultivar_df  <- data.frame(cultivar = each_cultivar, inf_value = each_cultivar_inf[3])
+  inflection_points <- rbind(each_cultivar_df, inflection_points)
+}
+
+all_cultivars <- left_join(all_cultivars, inflection_points, by = "cultivar")
+```
+
+```{r plot_inflections}
+ggplot(all_cultivars) +
+  geom_point(aes(x = gdd_cum, y = mean)) +
+  geom_line(aes(x = gdd_cum, y = mean_predict), color = "orange") +
+  geom_vline(aes(xintercept = inf_value)) +
+  facet_wrap(~cultivar, scales = "free_y") +
+  labs(x = "Cumulative growing degree days", y = "Canopy Height")
+
+ggplot(inflection_points, aes(x = inf_value)) +
+  geom_histogram() +
+  xlim(min(all_cultivars$gdd_cum), max(all_cultivars$gdd_cum)) +
+  labs(x = "Inflection points", y = "Number")
+```
+
+## Get image data
+
+For the second analysis, we are going to put together some trait data with RGB sensor data. 
+These data are for May 2018, and are read in below. 
+The image data are in the raster format. 
+
+```{r get_image_data}
+library(raster)
+library(rgdal)
+library(stringr)
+
+# File downloaded from: https://terraref.ncsa.illinois.edu/clowder/datasets/5c507cb74f0c436195ba8eb4?space=5c50512a4f0c436195b9ad67
+tif_url <- "vignettes/rgb_geotiff_L1_ua-mac_2018-05-04__13-04-09-888_left.tif"
+
+rgb_raster_1      <- raster(tif_url, band = 1)
+rgb_raster_1_crop <- crop(rgb_raster_1, extent(rgb_raster_1, 1500, 2500, 3000, 4000))
+plot(rgb_raster_1_crop)
+cellStats(rgb_raster_1_crop, stat = "mean")
+hist(rgb_raster_1_crop)
+
+rgb_raster_2      <- raster(tif_url, band = 2)
+rgb_raster_2_crop <- crop(rgb_raster_2, extent(rgb_raster_2, 1500, 2500, 3000, 4000))
+plot(rgb_raster_2_crop)
+cellStats(rgb_raster_2_crop, stat = "mean")
+hist(rgb_raster_2_crop)
+```
+
+```{r get_trait_data_2, message = FALSE}
+trait2 <- betydb_query(table     = "search", 
+                       trait     = "canopy_cover", 
+                       date      = "~2018 May",
+                       limit     =  "none")
+
+trait2 <- trait2 %>% 
+  mutate(day = as.Date(raw_date))
+```
+
+## Calculate greenness metric for each image
+
+We then want a summary statistic for the entire image data. 
+The chosen statistic here uses the red and green bands from the image raster. 
+This metric is a way to characterize the greenness of the area where the data were taken from. 
+
+```{r image_calculations}
+add_rasters <- rgb_raster_2_crop + rgb_raster_1_crop
```

Comment:
it isn't clear here where / how the NGDRI equation is implemented - I assume that the `_1_` and `_2_` refer to the red and green layers? It would be nice to have object names like red and green (?) that represent the different layers and then have the equation like 

```r
ngdri <- (red - green) / (red + green)
```